photo-helper/src/main/java/io/intrepid/photohelper/PhotoHelper.java

```diff
@@ -0,0 +1,169 @@
+    @VisibleForTesting
+    static final String CAMERA_IMAGE_FILE_PATH = "camera_image_file_path";
+    @VisibleForTesting
+    static final String REMOVE_UNUSED_PHOTOS = "remove_unused_photos";
+    @VisibleForTesting
+    static final int REQUEST_CODE_PICK_PHOTO = 31;
+    @VisibleForTesting
+    static final int REQUEST_CODE_WRITE_STORAGE_PERMISSION = 45;
+
+    private Fragment fragment;
+    private PhotoContract.View view;
+    @VisibleForTesting
+    private boolean removeUnusedPhotos;
+    String cameraImageFilePath;
+
+    /**
+     * Creates a PhotoHelper instance
+     *
+     * @param fragment           - The {@link Fragment} that will be calling the {@link PhotoHelper}
+     * @param view               - The {@link io.intrepid.photohelper.PhotoContract.View View} interface (may very likely be the {@link Fragment})
+     * @param removeUnusedPhotos - If true, delete unused new photos that the user takes with the camera
+     */
+    public PhotoHelper(@NonNull Fragment fragment, @NonNull PhotoContract.View view, boolean removeUnusedPhotos) {
+        this.fragment = fragment;
+        this.view = view;
+        this.removeUnusedPhotos = removeUnusedPhotos;
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        if (resultCode == Activity.RESULT_OK) {
+            if (requestCode == REQUEST_CODE_PICK_PHOTO) {
+                if (data != null && data.getData() != null) {
+                    Uri imageUri = data.getData();
+                    // User picked the picture from gallery
+                    view.getPhotoPresenter().onPictureChosen(imageUri.toString());
+                    deleteUnusedPhotos();
+                    return;
+                } else {
+                    if (cameraImageFilePath != null) {
+                        // User took a new photo
+                        view.getPhotoPresenter().onPictureChosen(cameraImageFilePath);
+
+                        // Reset this for next time
+                        cameraImageFilePath = null;
+                        return;
+                    }
+                }
+            }
+        }
+
+        view.getPhotoPresenter().onNoPictureChosen();
+        deleteUnusedPhotos();
+    }
+
+    /**
+     * Helper function to delete unused photos
+     */
+    @VisibleForTesting
+    void deleteUnusedPhotos() {
+        if (removeUnusedPhotos && cameraImageFilePath != null) {
+            File file = new File(cameraImageFilePath);
+            //noinspection ResultOfMethodCallIgnored
+            file.delete();
+        }
+        cameraImageFilePath = null;
+
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        if (savedInstanceState != null) {
+            cameraImageFilePath = savedInstanceState.getString(CAMERA_IMAGE_FILE_PATH);
+            removeUnusedPhotos = savedInstanceState.getBoolean(REMOVE_UNUSED_PHOTOS);
```

Comment:
is this necessary? if onCreate is being called, presumably a new photohelper object is created, in which case the user would just pass in removeUnusedPhotos to the constructor